include/dbcppp/Iterator.h

```diff
@@ -1,102 +1,117 @@
         }
-        self_t& operator-=(std::size_t o)
+        constexpr self_t& operator-=(std::size_t o) noexcept
         {
             _i -= o;
             return *this;
         }
 
-        bool operator==(const self_t& rhs) const
+        constexpr bool operator==(const self_t& rhs) const noexcept
         {
             return _i == rhs._i;
         }
-        bool operator!=(const self_t& rhs) const
+        constexpr bool operator!=(const self_t& rhs) const noexcept
         {
             return !(*this == rhs);
         }
 
     private:
-        get_t _get;
+        P _parent;
+        F _get;
         std::size_t _i;
     };
-    template <class Iterator>
-    class Iterable
+    template<class T, class P, typename F>
+    class Iterable final
     {
     public:
-        Iterable(Iterator begin, Iterator end)
-            : _begin(begin)
-            , _end(end)
+        using value_type = const T;
+        using reference = value_type&;
+        using const_reference = const value_type&;
+        using iterator = Iterator<T, P, F>;
+        using const_iterator = Iterator<const T, P, F>;
+        using difference_type = std::ptrdiff_t;
+        using size_type = std::size_t;
+
+        constexpr Iterable(P parent, F get, size_t size) noexcept
+            : _parent(std::move(parent))
+            , _get(std::move(get))
+            , _size(std::move(size))
         {}
-        Iterator& begin()
+        constexpr iterator begin() noexcept
+        {
+            return iterator(_parent, _get, 0);
+        }
+        constexpr iterator end() noexcept
+        {
+            return iterator(_parent, _get, _size);
+        }
+        constexpr const_iterator begin() const noexcept
         {
-            return _begin;
+            return const_iterator(_parent, _get, 0);
         }
-        Iterator& end()
+        constexpr const_iterator end() const noexcept
         {
-            return _end;
+            return const_iterator(_parent, _get, _size);
         }
 
     private:
-        Iterator _begin;
-        Iterator _end;
+        P _parent;
+        F _get;
+        size_t _size;
     };
 }
-#define DBCPPP_MAKE_ITERABLE(ClassName, Name, Type)                                 \
-    auto Name() const                                                               \
-    {                                                                               \
-        auto get = std::bind(&ClassName::Name##_Get, this, std::placeholders::_1);  \
-        Iterator<Type> begin(get, 0);                                               \
-        Iterator<Type> end(get, Name##_Size());                                     \
-        return Iterable(begin, end);                                                \
+#define DBCPPP_MAKE_ITERABLE(ClassName, Name, Type)                                                       \
+    inline Iterable<const Type, const ClassName*, decltype(&ClassName::Name##_Get)> Name() const noexcept \
```

Comment:
Ich glaube hier sollte noch explizit ein `DBCPPP_NO_EXPORT` bei stehen.

Reply:
... doesn't matter. Correctly setting up the compiler flags regarding symbol visibility of inlines (only applicable to GNU platforms in the first place) isn't of concern to this library. Just making it `inline` is enough to have the correct effect.